django/contrib/admin/static/admin/js/inlines.js

```diff
@@ -296,7 +296,14 @@
                     dependency_list = input.data('dependency_list') || [],
                     dependencies = [];
                 $.each(dependency_list, function(i, field_name) {
-                    dependencies.push('#' + row.find('.form-row .field-' + field_name).find('input, select, textarea').attr('id'));
+                    let field_element = row.find('.form-row.field-' + field_name + ' .field-' + field_name).find('input, select, textarea');
```

Comment:
Can you explain this change? This is tricky

`'.form-row .field-' + field_name`
->
`'.form-row.field-' + field_name + ' .field-' + field_name`

Reply:
The below screenshot is for prepopulated fields in stacked inlines with fieldsets (`title` and `pubdate` are dependent fields for slug1). So we can access the title field by using the class name `.form-row.field-title .field-title`.

![Screenshot from 2021-11-16 12-31-55](https://user-images.githubusercontent.com/59529068/141942587-788f1652-28ad-492d-a80d-5afd17b9484a.png)


The below screenshot is for prepopulated fields in stacked inlines without fieldsets (`title` is the dependent field for slug1). So we can access the title field by using the class name `.form-row.field-title`.

![Screenshot from 2021-11-16 12-32-21](https://user-images.githubusercontent.com/59529068/141942829-37cbbaa2-7510-4344-af9a-ec642f91a909.png)

Now, what I did was, try to find the dependent field assuming that it is in a fieldset hence 
`field_element = row.find('.form-row.field-title .field-title').find('input, select, textarea')` (assume that the field_name is title). If the field is really in the fieldset then the `row.find` method would return me that element otherwise it returns me an object with length zero. If the length property of the returned object is zero then it means that the field is not inside a fieldset, in that case, field_element would become
`row.find('.form-row .field-title').find('input, select, textarea')` (assume that field_name is title)

I hope I was able to make things clear.

Reply:
Thanks for extra details. I think it will be more readable to leave the current selector and add a comment, e.g.
```js
// Dependency in a fieldset.
let field_element = row.find('.form-row .field-' + field_name);
// Dependency without a fieldset.
if (!field_element.length) {
    field_element = row.find('.form-row.field-' + field_name);
}
dependencies.push('#' + field_element.find('input, select, textarea').attr('id'));
```